### PR TITLE
Make SessionWrapper related fields immutable that are part of the equals method

### LINKED ISSUE #28906: ID fields in SessionWrapper should be immutable
### Description

When analyzing issue #28466, I did some research to ensure to understand that the equal method of SessionWrapper and the replacing of values in the embedded cached in InfinispanChangelogBasedTransaction.

I found that the logic relies on that the version of a session wrapper and the ID of the entity are never changed, as with embedded caches Infinispan would return the object if it is local, and a modification on those items would then change the local cache which would result in failing to replace the entity later. 

This enhancement will put that into code and makes those fields final, so it won't break in the future.

### Discussion

_No response_

### Motivation

Make the code safe so it won't break by accident.

### Details

I'll prepare a PR.

### PR DESCRIPTION
**Pull request by ahus1**
The cache replace logic depends on it, as values returned by reference from a local cache must never be modified on those critical fields directly.

Closes #28906

### CODE REVIEW

**Comment by ahus1** [comment]
@pruivo - FYI, this is an attempt to make the code safer to use. It didn't reveal any problems, and this code change is only there to protect it against future problems.

**Comment by ahus1** [comment]
@mposolda - as you have authored most of the code, I wonder if you could please give it a review and add your thoughts. Thanks!

**Review by mhajas — APPROVED** [review]
LGTM @ahus1! I am approving, I will leave it to you whether you want to wait for other reviews.

**Review by pruivo — APPROVED** [review]

**Pull request merged**
